### PR TITLE
Add PlayerInteractEntityEvent

### PR DESCRIPTION
**Pull request by pagoru**
This code implements the player entity event.
See https://github.com/SpongePowered/SpongeCommon/issues/62

### CODE REVIEW

File: src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java [hunk 1]
```diff
@@ -347,5 +350,32 @@ public void onDisconnectPlayer(IChatComponent reason, CallbackInfo ci) {
         Sponge.getGame().getEventManager().post(event);
         event.getSink().sendMessage(event.getNewMessage());
     }
+    
+    @Inject(method = "processUseEntity", at = @At(value = "INVOKE", target =
+             "net/minecraft/network/play/client/C02PacketUseEntity.getAction()Lnet/minecraft/network/play/client/C02PacketUseEntity$Action;",
+             ordinal = 0), cancellable = true, locals = LocalCapture.CAPTURE_FAILSOFT)
+    public void processUseEntity(C02PacketUseEntity packetIn, CallbackInfo info, WorldServer worldserver, Entity entity, boolean flag, double d0) {
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java]: I believe the `boolean flag` parameter needs the `@Coerce` annotation. `double` is fine I think

File: src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java [hunk 2]
```diff
@@ -347,5 +350,32 @@ public void onDisconnectPlayer(IChatComponent reason, CallbackInfo ci) {
...
+        EntityInteractionType type = null;
+        Vector3d location = null;
+        switch (packetIn.getAction()) {
+            case ATTACK:
+                type = EntityInteractionTypes.ATTACK;
+                break;
+            case INTERACT_AT:
+                location = new Vector3d(packetIn.getHitVec().xCoord, packetIn.getHitVec().yCoord, packetIn.getHitVec().zCoord);
```
Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java]: `location = VecHelper.toVector(packetIn.getHitVec());` is neater

Comment (simon816) [inline src/main/java/org/spongepowered/common/mixin/core/network/MixinNetHandlerPlayServer.java]: Also, you're not setting the `type` variable in this block, should be `EntityInteractionTypes.USE`

**Comment by Zidane** [comment]
@simon816 

Biggest issue I see with this PR (besides what you've pointed out) is the fact that Forge already has an event for this (pretty sure it does) and therefore this really should go into SpongeVanilla (and mixed into the Forge event in Sponge)

**Comment by pagoru** [comment]
I fucked up the commit... D=
I need remember the "branches"...

**Pull request closed without merging**